Sources/Auth/Internal/SessionManager.swift

```diff
@@ -55,40 +55,50 @@ private actor LiveSessionManager {
 
   func refreshSession(_ refreshToken: String) async throws -> Session {
     try await SupabaseLoggerTaskLocal.$additionalContext.withValue(
-      merging: ["refreshID": .string(UUID().uuidString)]
+      merging: ["refresh_id": .string(UUID().uuidString)]
     ) {
       try await trace(using: logger) {
         if let inFlightRefreshTask {
-          logger?.debug("refresh already in flight")
+          logger?.debug("Refresh already in flight")
           return try await inFlightRefreshTask.value
         }
 
         inFlightRefreshTask = Task {
-          logger?.debug("refresh task started")
+          logger?.debug("Refresh task started")
 
           defer {
             inFlightRefreshTask = nil
-            logger?.debug("refresh task ended")
+            logger?.debug("Refresh task ended")
           }
 
-          let session = try await api.execute(
-            HTTPRequest(
-              url: configuration.url.appendingPathComponent("token"),
-              method: .post,
-              query: [
-                URLQueryItem(name: "grant_type", value: "refresh_token"),
-              ],
-              body: configuration.encoder.encode(UserCredentials(refreshToken: refreshToken))
+          do {
+            let session = try await api.execute(
+              HTTPRequest(
+                url: configuration.url.appendingPathComponent("token"),
+                method: .post,
+                query: [
+                  URLQueryItem(name: "grant_type", value: "refresh_token"),
+                ],
+                body: configuration.encoder.encode(
+                  UserCredentials(refreshToken: refreshToken)
+                )
+              )
             )
-          )
-          .decoded(as: Session.self, decoder: configuration.decoder)
+            .decoded(as: Session.self, decoder: configuration.decoder)
 
-          update(session)
-          eventEmitter.emit(.tokenRefreshed, session: session)
+            update(session)
+            eventEmitter.emit(.tokenRefreshed, session: session)
 
-          await scheduleNextTokenRefresh(session)
+            await scheduleNextTokenRefresh(session)
 
-          return session
+            return session
+          } catch {
+            logger?.debug("Failed to refresh token: \(error)")
+            remove()
+            eventEmitter.emit(.signedOut, session: nil)
+
+            throw error
```

Comment:
The HTTP client already handles request retries, so by this point token refresh was already retried a couple of times, we can remove it right away.